controllers/oauth2client_controller.go

```diff
@@ -352,17 +364,27 @@ func parseSecret(secret apiv1.Secret, authMethod hydrav1alpha1.TokenEndpointAuth
 	}, nil
 }
 
-func (r *OAuth2ClientReconciler) getHydraClientForClient(oauth2client hydrav1alpha1.OAuth2Client) (HydraClientInterface, error) {
+func (r *OAuth2ClientReconciler) getHydraClientForClient(
```

Comment:
This IMO should have a test, unit at least

Reply:
I'll work on this today.

Reply:
Usually I find it a bit anti-idiomatic to create a unit test for a private function.

I'm wondering if we should we create an integration test for this instead?